SF-3092 Fix draft notices displaying when drafting is not enabled.

kylebuss opened the pull request:
This PR removes app notices from the Drafting page when projects are not setup for drafting. New tests were added to the `draft-gerenation.component.spec.ts` to verify notices either appear or do not appear depending on if project `isPreTranslationApproved`.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/sillsdev/web-xforge/2864)
<!-- Reviewable:end -->

RaymondLuong3 reviewed (changes requested):
Nice work. I few comments to look at.

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2864#-:-OCdYuXdCl404VZewEip:bxmegnn)* status: all files reviewed, 3 unresolved discussions (waiting on @kylebuss)

___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/translate/draft-generation/draft-generation.component.spec.ts` line 345 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2864#-OCdYL5qAwJMZgFX-hQ6:-OCdYL5qAwJMZgFX-hQ7:bcmparu) ([raw file](https://github.com/sillsdev/web-xforge/blob/a76931251c06091763a3c704b5444f4c6f6e4049/src/SIL.XForge.Scripture/ClientApp/src/app/translate/draft-generation/draft-generation.component.spec.ts#L345)):*
> ```XML
>     }));
> 
>     it('should detect source language same as target language but not display warning message if drafting is not enabled', fakeAsync(() => {
> ```

Since this is a test for back translations, the expectation will then be that the warnings are present when languages are the same. You will want to duplicate this test for the case where a project is not a back translation and not approved and show that the warning appears.



_Code quote:_
```XML
    it('should detect source language same as target language but not display warning message if drafting is not enabled', fakeAsync(() => {
```
___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/translate/draft-generation/draft-generation.component.html` line 10 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2864#-OCdWoLT6g2SzjHr5wKv:-OCdWoLT6g2SzjHr5wKw:bm0k96) ([raw file](https://github.com/sillsdev/web-xforge/blob/a76931251c06091763a3c704b5444f4c6f6e4049/src/SIL.XForge.Scripture/ClientApp/src/app/translate/draft-generation/draft-generation.component.html#L10)):*
> ```HTML
>       </h1>
>     }
>     @if (isPreTranslationApproved) {
> ```

Back translation projects are automatically allowed to draft even without approval. You may want to add a check for `isBackTranslationMode` here.



_Code quote:_
```HTML
    @if (isPreTranslationApproved) {
```
___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/translate/draft-generation/draft-generation.component.html` line 93 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2864#-OCdXayiDCEMwnrOY9Dz:-OCdXayiDCEMwnrOY9E-:byc9mqa) ([raw file](https://github.com/sillsdev/web-xforge/blob/a76931251c06091763a3c704b5444f4c6f6e4049/src/SIL.XForge.Scripture/ClientApp/src/app/translate/draft-generation/draft-generation.component.html#L93)):*
> ```HTML
> 
>       <!-- Only show warnings if target language is supported -->
>       @if (isPreTranslationApproved && isTargetLanguageSupported) {
> ```

We should show warnings if this is in back translation mode and not approved for pre-translation.



_Code quote:_
```HTML
      @if (isPreTranslationApproved && isTargetLanguageSupported) {
```


<!-- Sent from Reviewable.io -->

kylebuss commented:
*[`src/SIL.XForge.Scripture/ClientApp/src/app/translate/draft-generation/draft-generation.component.spec.ts` line 345 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2864#-OCdYL5qAwJMZgFX-hQ6:-OCeKbehCuJzou6CZnGO:b-cdf3j0) ([raw file](https://github.com/sillsdev/web-xforge/blob/a76931251c06091763a3c704b5444f4c6f6e4049/src/SIL.XForge.Scripture/ClientApp/src/app/translate/draft-generation/draft-generation.component.spec.ts#L345)):*
<details><summary><i>Previously, RaymondLuong3 (Raymond Luong) wrote…</i></summary><blockquote>

Since this is a test for back translations, the expectation will then be that the warnings are present when languages are the same. You will want to duplicate this test for the case where a project is not a back translation and not approved and show that the warning appears.
</blockquote></details>

Updated this to check the draft speed notice and added tests in the warnings section to check show/hide of messages.


<!-- Sent from Reviewable.io -->

kylebuss reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2864)* status: 0 of 2 files reviewed, 3 unresolved discussions (waiting on @RaymondLuong3)

___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/translate/draft-generation/draft-generation.component.html` line 10 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2864#-OCdWoLT6g2SzjHr5wKv:-OCeKaXo1bE4OAPcDIlt:b-896fix) ([raw file](https://github.com/sillsdev/web-xforge/blob/a76931251c06091763a3c704b5444f4c6f6e4049/src/SIL.XForge.Scripture/ClientApp/src/app/translate/draft-generation/draft-generation.component.html#L10)):*
<details><summary><i>Previously, RaymondLuong3 (Raymond Luong) wrote…</i></summary><blockquote>

Back translation projects are automatically allowed to draft even without approval. You may want to add a check for `isBackTranslationMode` here.
</blockquote></details>

Done.
___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/translate/draft-generation/draft-generation.component.html` line 93 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2864#-OCdXayiDCEMwnrOY9Dz:-OCeKb-Y4HqzbbPrPDfH:b-896fix) ([raw file](https://github.com/sillsdev/web-xforge/blob/a76931251c06091763a3c704b5444f4c6f6e4049/src/SIL.XForge.Scripture/ClientApp/src/app/translate/draft-generation/draft-generation.component.html#L93)):*
<details><summary><i>Previously, RaymondLuong3 (Raymond Luong) wrote…</i></summary><blockquote>

We should show warnings if this is in back translation mode and not approved for pre-translation.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

kylebuss reviewed (commented):
@Nateowami Done.

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2864#-:-OCi7iI--Ts-7bjNprtu:bu33uki)* status: 0 of 3 files reviewed, 3 unresolved discussions (waiting on @RaymondLuong3)



<!-- Sent from Reviewable.io -->

RaymondLuong3 reviewed (approved):
We should probably remove the allow forward NMT translation feature flag at some point. But that would be another task. Nice work!

> Reviewed 1 of 2 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2864#-:-OCilGgt300Y-wDEU7lL:b-oqe5jo)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @kylebuss)



<!-- Sent from Reviewable.io -->
